### PR TITLE
Adds a Quartermaster class to manage machines and materiel

### PR DESCRIPTION
**Pull request by sixlettervariables**
Does what it says, centralizing adding, arriving, buying, etc. Slims down Campaign a lot, and enables us to slim down `Part::fix` and `Part::remove` in a future PR.

As always, lots of new tests.

### CODE REVIEW

**Comment by sixlettervariables** [comment]
Okay this adds close to 100% coverage of the new Quartermaster class and the existing methods copied over.

**Review by Windchild292 — COMMENTED** [review]
One comment and a few formatting recommendations, otherwise this looks good.

File: MekHQ/src/mekhq/campaign/parts/AmmoStorage.java [hunk 1]
```diff
@@ -61,8 +58,8 @@ public AmmoStorage() {
     public AmmoStorage(int tonnage, EquipmentType et, int shots, Campaign c) {
         super(tonnage, et, -1, 1.0, c);
         this.shots = shots;
-        if(null != type && type instanceof AmmoType) {
-        	this.munition = ((AmmoType)type).getMunitionType();
+        if (type instanceof AmmoType) {
+        	this.munition = ((AmmoType) type).getMunitionType();
```
[the remark this answers is not in the document]
Comment (sixlettervariables) [reply MekHQ/src/mekhq/campaign/parts/AmmoStorage.java:62]: Not sure how that didn't get replaced!

**Comment by sixlettervariables** [comment]
CI failure is a unit test failure post merge. Will try to look into this today.

**Review by Windchild292 — APPROVED** [review]
Approved pending CI

**Pull request merged**